internal/sharedcomponent/sharedcomponent.go

```diff
@@ -12,23 +13,25 @@ import (
 	"go.opentelemetry.io/collector/component"
 )
 
-// SharedComponents a map that keeps reference of all created instances for a given configuration,
-// and ensures that the shared state is started and stopped only once.
-type SharedComponents[K comparable, V component.Component] struct {
-	comps map[K]*SharedComponent[V]
+// Map keeps reference of all created instances for a given shared key such as a component configuration.
+type Map[K comparable, V component.Component] interface {
```

Comment:
Why interface? We don't expect others to implement or have alternative implementations.

Reply:
Going overboard, thanks for feedback, will remove.